pkg/dns/service.go

```diff
@@ -149,53 +130,78 @@ func (s *Service) GetDNSRecord(ctx context.Context, subDomain string, managedZon
 	return dnsRecord, nil
 }
 
-// SetEndpoints sets the endpoints for the given DNSRecord for each of the gateway addresses.
-func (s *Service) SetEndpoints(ctx context.Context, addresses []gatewayv1beta1.GatewayAddress, dnsRecord *v1alpha1.DNSRecord, dnsPolicy *v1alpha1.DNSPolicy) error {
-	//dnsPolicy is not currently used, but will be once we add other routing strategies
-	ips, err := s.resolveIPS(ctx, addresses)
-	if err != nil {
-		return err
-	}
+// SetEndpoints sets the endpoints for the given MultiClusterGatewayTarget
+func (s *Service) SetEndpoints(ctx context.Context, mcgTarget *MultiClusterGatewayTarget, dnsRecord *v1alpha1.DNSRecord) error {
```

Comment:
this is quite a complex function now. might be worth adding a bit more description on what it is doing?

Reply:
Updated, hopefully it makes sense